libraries/AP_ExternalAHRS/AP_ExternalAHRS_VectorNav.cpp

```diff
@@ -625,11 +518,7 @@ void AP_ExternalAHRS_VectorNav::process_ahrs_packet(const uint8_t *b)
 #if AP_COMPASS_EXTERNALAHRS_ENABLED
     {
         AP_ExternalAHRS::mag_data_message_t mag;
-        if (use_uncomp) {
```

Comment:
This existing functionality has been removed?

Reply:
While the naming of uncompaccel/uncompgyro/uncompmag implies they're handled the same, a distinct difference is the mag (compensated) isn't estimated by the internal EKF; our on-board Hard Iron estimator is a pretty rarely used feature, so it seemed like unnecessary bloat and bytes to me (especially given we're using it in our high-rate message). If we think that's still valuable I can readd it.

Reply:
It is not only removed for mag? Its removed for accel/gyro too?

Reply:
It's available for accel/gyro, [here](https://github.com/ArduPilot/ardupilot/pull/27585/files#diff-d01800740fba47f0cbbd7c65d64588ac01381d0464bd10075b00e46ab0149cefL602)

Reply:
Ah, OK. So is the mag compensated or not now? My preference would be uncompensated I think. Maybe it is possible to turn off the compensation on the VN side so the user can still pick if they really care?

My argument for uncompensated is that AP should be able to do a better job of calibration as it can use battery current if it needs to where as the VN does not have that information.

Reply:
It's technically using compensated, but unless the user has explicitly configured the VN to run the HSI estimator, the compensated and uncompensated outputs are both uncompensated.